### DISCUSSION TITLE
How does reset work exactly?

### DISCUSSION THREAD

**Discussion by Geddard** [react-hook-form/react-hook-form#3886] [Q&A]
Hi there, I have just a quick question about reset, which I didn't get from the API docs.

The first param for reset, does it have to be the whole defaultValues object?
Or if, for example, I have a form with 10 fields, and I just want to reset two of them, can I just send those two fields with the "reset" values?

Example:

```
const defaultValues = {
  firstName: "",
  lastName: "",
  height: "",
  age: "",
}

// Let's assume there is a form for all of these with inputs, and the input values are currently, respectively: "John", "Smith", "180", "22"

// So now I want to reset only height and age, and leave the other fields as dirty, can I do this? ->

reset({
 height: "",
 age: ""
}, { dirtyFields: true});


``` 
What I hope to achieve here is to keep first and last name with the "dirty" values and also the formState.dirtyFields with those two values only

My only other idea would be to loop through the field names and setValue them to empty with `shouldDirth: false`

Thanks for the help!

**Comment by bluebill1049** [comment]
It's best to provide the entire `defaultValues`.

**Reply by Geddard** [reply]
> It's best to provide the entire `defaultValues`.

So, is there a nice way of resetting only some fields and remove the dirt from them? but keep the dirt on others?

**Reply by bluebill1049** [reply]
```
reset({
   ...defaultValues,
   ...youFields?
}, { dirtyFields: true});
```

we don't have a specific API for such action, also reset API is getting some improvement in the next version. right now it's a bit confusing.
